agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/MetricDataMapper.java

```diff
@@ -170,6 +172,10 @@ public static void updateMetricPointBuilder(
       boolean success = isSuccess(statusCode, captureHttpServer4xxAsError);
       Boolean isSynthetic = attributes.get(IS_SYNTHETIC);
       if (metricData.getName().contains(".server.")) {
+        attributes.forEach(
+            (key, value) ->
+                SpanDataMapper.applyConnectionStringAndRoleNameOverrides(
+                    metricTelemetryBuilder, value, key.getKey()));
```

Comment:
this is applied to all cases.. can you move it to the beginning so that it's applied only once for all cases?
setExtraAttributes can exclude it.

Reply:
done